### ISSUE TITLE
2FA in firefox extension & android app

### ISSUE THREAD

**Issue by aposorio** [bitwarden/browser#229]
First, thanks kindly for all your hard work; bitwarden is my first experience with a password manager and thus far the experience has been good.

I enabled 2FA for the web vault and it appears to be working as expected. However, if I lock or log out of the Firefox extension, I am not prompted to provide the pin from my authenticator app. The same is true with the Android app.

I may be missing some basic point but this seems to be a rather large loophole in terms of defeating the extra security which 2FA is intended to provide. For example, if someone were to obtain my master password, they would _not_ be able to access my web vault without also having access to my device with the authenticator app. However, the same person _would_ be able to access my vault as long as they used the Android app or the Firefox extension--I'm not sure whether the same applies to other extensions or the iOS app--since they would not be required to provide that second layer of authentication.

I appreciate your time and attention and I apologize if I've raised a non-issue.

P.S. I saw that someone raised what appears to be a similar issue with regards to the Chrome extension back in November of 2016: https://github.com/bitwarden/browser/issues/51

**Comment by kspearrin** [comment]
Hi,
This is likely happening because:

1. You are not fully logging out and back in. 2FA only occurs when logging in, not when unlocking the app.

2. You logged in previously and selected "Remember Me" on the 2FA step. This skips 2FA on that device for 30 days or until you deauthorize all sessions from the web vault.

**Comment by aposorio** [comment]
Confirmed. I deauthorized all sessions from the web vault and when I attempted to log in through the Firefox extension or the Android app I was prompted to enter my 2FA code. Thanks kindly for following up on this. 

I did have a related question: Even after deauthorizing the sessions, the field for my email/username was already filled in on the Android app and the Firefox extension offered to autocomplete it after typing the first letter. Is that by design? Is there a way to ensure that all login information is removed after deauthorizing sessions?

Thanks again.
